### PR TITLE
Log block execution latency metric

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

Implementing metrics in https://docs.google.com/spreadsheets/d/1tHo8iKzrlkuYeN2Yu7a6MfwQd-P2KNA9BlnhcASoEQY/edit?pli=1#gid=0

## Proposal

Implementing block execution latency Histogram metric

## Test Plan

Ran validator locally, ran a benchmark against it, and saw metric properly logged:

With this one, since it's a Histogram, you can do more stuff, like get the p99:

For this benchmark, the p99 for block execution latency was 4.95 milliseconds

### CODE REVIEW

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1132** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1132" target="_blank">Graphite</a> 
    * **PR #1133** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1133" target="_blank">Graphite</a> 
      * **PR #1134** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1134" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1134?utm_source=stack-comment).

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1132** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1132" target="_blank">Graphite</a> 
    * **PR #1133** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1133" target="_blank">Graphite</a> 
      * **PR #1134** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1134" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1134?utm_source=stack-comment).

File: linera-chain/src/chain.rs [hunk 1]
```diff
@@ -46,6 +49,15 @@ pub static NUM_BLOCKS_EXECUTED: Lazy<IntCounterVec> = Lazy::new(|| {
     )
     .expect("Counter can be created")
 });
+pub static BLOCK_EXECUTION_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
+        "block_execution_latency",
+        "Block execution latency",
+        // Can add labels here
+        &[]
```
Comment (andresilva91) [inline linera-chain/src/chain.rs:57]: This is where I mentioned we can add labels like `chain_id` or whatever else we want

File: linera-chain/src/chain.rs [hunk 2]
```diff
@@ -541,6 +555,9 @@ where
 
         // Log Prometheus metrics
         NUM_BLOCKS_EXECUTED.with_label_values(&[]).inc();
+        BLOCK_EXECUTION_LATENCY
+            .with_label_values(&[])
```
Comment (andresilva91) [inline linera-chain/src/chain.rs:559]: Then we can log the label values here

**Review by ma2bd — APPROVED** [review]

**Comment by andresilva91** [comment]
### Merge activity

* **Oct 16, 11:20 PM**: @andresilva91 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1134).

**Pull request merged**